mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -692,14 +692,20 @@ private[spark] object RandomForest extends Logging {
       node.stats
     }
 
-    // For each (feature, split), calculate the gain, and select the best (feature, split).
-    val (bestSplit, bestSplitStats) =
-      Range(0, binAggregates.metadata.numFeaturesPerNode).map { featureIndexIdx =>
-        val featureIndex = if (featuresForNode.nonEmpty) {
-          featuresForNode.get.apply(featureIndexIdx)
+    val validFeatureSplits =
+      Range(0, binAggregates.metadata.numFeaturesPerNode).view.map { featureIndexIdx =>
+        if (featuresForNode.nonEmpty) {
+          (featureIndexIdx, featuresForNode.get.apply(featureIndexIdx))
```

Comment:
I don't think so. The alternative is `featuresForNode.get(featureIndexIdx)` which is misleading even though it does work. It looks like you are calling a function `get` and passing `featureIndexIdx` as an argument. Explicit apply seems clearer.